### ISSUE TITLE
Style/CombinableLoops incorrectly triggers on rails validates_each

### ISSUE THREAD

**Issue by TikiTDO** [rubocop-hq/rubocop#8848]
The `Style/CombinableLoops` cop assumes that any method that ends with `_each` is an iterator, and then applies the cop logic to the method body. This rule triggers on the rails `validates_each` method, described [here](https://apidock.com/rails/ActiveModel/Validations/ClassMethods/validates_each), which is not actually an iterator.

The below code will trigger the cop:

```ruby
class Example < ActiveRecord::Base
  # Has two attributes :array_values, :other_array_values, both of which are arrays of strings

  validates_each :array_values, :other_array_values do |record, attribute, array_value|
    array_value.each do |array_value_item|
      record.errors.add(attribute, "invalid") unless is_valid_item?(array_value_item)
    end
  end

  def is_valid_item?(item)
    return item !== "invalid"
  end
end
```

--------

## Expected behavior

The `Style/CombinableLoops` should not necessarily trigger on methods with names that happen to end with `_each`, since that suffix doesn't necessarily imply that it's an iterator. 

## Actual behavior

The above code is not actually a combinable loop, because the `validates_each` is not really an iterator, but simply a method that can apply some validation logic to a value which may in turn be an enumerable. As such it should not trigger this cop.

## Steps to reproduce the problem

See example above in the issue description above. There is no way to combine `validates_each` and the iteration over the `array_value` array.

## RuboCop version

```
└> bundle exec rubocop -V
0.92.0 (using Parser 2.7.1.5, rubocop-ast 0.7.1, running on ruby 2.7.1 x86_64-linux)
```

**Comment by dvandersluis** [comment]
`Style/CombinableLoops` was changed in #8731 which seems to have fixed this bug, I can't reproduce this error.

**Comment by dvandersluis** [comment]
Actually I did find a case that this does register incorrectly, writing a PR now.
